mne_connectivity/base.py

```diff
@@ -709,11 +714,15 @@ def get_data(self, output="compact"):
         if output == "raveled":
             data = self._data
         else:
-            if self.method in ["cacoh", "mic", "mim", "gc", "gc_tr"]:
-                # multivariate results cannot be returned in a dense form as a
-                # single set of results would correspond to multiple entries in
-                # the matrix, and there could also be cases where multiple
-                # results correspond to the same entries in the matrix.
+            if (
+                isinstance(self.indices, tuple)
+                and not np.all(isinstance(self.indices[0], int))
+                and not np.all(isinstance(self.indices[1], int))
+            ):  # i.e. check if multivariate results based on nested indices
+                # multivariate results cannot be returned in a dense form as a single
+                # set of results would correspond to multiple entries in the matrix, and
+                # there could also be cases where multiple results correspond to the
+                # same entries in the matrix.
                 raise ValueError(
                     "cannot return multivariate connectivity data in a dense form"
                 )
```

Comment:
When updating the container unit tests I realised using the indices format is more a consistent check for multivariate conn scores rather than the method name